certbot/certbot/_internal/snap_config.py

```diff
@@ -0,0 +1,94 @@
+"""Module configuring Certbot in a snap environment"""
+import socket
+import sys
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+try:
+    from urllib3.connection import HTTPConnection
+    from urllib3.connectionpool import HTTPConnectionPool
+except ImportError:
+    # Stub imports for oldest requirements, that will never be used in snaps.
+    HTTPConnection = object
+    HTTPConnectionPool = object
+
+from acme.magic_typing import List
+
+from certbot.compat import os
+
+
+_ARCH_TRIPLET_MAP = {
+    'arm64': 'aarch64-linux-gnu',
+    'armhf': 'arm-linux-gnueabihf',
+    'i386': 'i386-linux-gnu',
+    'ppc64el': 'powerpc64le-linux-gnu',
+    'powerpc': 'powerpc-linux-gnu',
+    'amd64': 'x86_64-linux-gnu',
+    's390x': 's390x-linux-gnu',
+}
+
+
+def prepare_env(cli_args):
+    # type: (List[str]) -> List[str]
+    """
+    Prepare runtime environment for a certbot execution in snap.
+    :param list cli_args: List of command line arguments
+    :return: Update list of command line arguments
+    :rtype: list
+    """
+    snap_arch = os.environ.get('SNAP_ARCH')
+
+    if snap_arch not in _ARCH_TRIPLET_MAP:
+        sys.stderr.write('Unrecognized value of SNAP_ARCH: {0}\n'.format(snap_arch))
+        sys.exit(1)
+
+    os.environ['CERTBOT_AUGEAS_PATH'] = '{0}/usr/lib/{1}/libaugeas.so.0'.format(
+        os.environ.get('SNAP'), _ARCH_TRIPLET_MAP[snap_arch])
+
+    session = Session()
+    session.mount('http://snapd/', _SnapdAdapter())
+
+    response = session.get('http://snapd/v2/connections?snap=certbot&interface=content')
+
+    if response.status_code == 404:
+        sys.stderr.write('An error occurred while fetching Certbot snap plugins: '
+                         'your version of snapd is outdated.\n')
+        sys.stderr.write('Please run "sudo snap install core" in your terminal and try again.\n')
```

Comment:
A small edge case:

One of the scenarios that Brad encountered was an Ubuntu AMI where `core` was installed at a very old version. snapd autorefresh had not yet had a chance to update it, because the server is freshly created.

AFAIK, `snap install` does not imply `snap refresh`. To close the loop here, we could add `; sudo snap refresh core`.

There's also the choice to do nothing, because the output from `snap install core` leaves a signpost for the user:

    snap "core" is already installed, see 'snap help refresh'

Reply:
I added your suggestion. It avoids one step for the user to solve the situation.